src/main/java/org/broadinstitute/hellbender/utils/io/IOUtils.java

```diff
@@ -646,11 +646,11 @@ public static File createFifoFile(final Path fifoFilePath, final boolean overwri
     /**
      * Makes a print stream for a file, gzipping on the fly if the file's name ends with '.gz'.
      */
-    public static PrintStream makePrintStreamMaybeGzipped(File file) throws IOException {
-        if (file.getPath().endsWith(".gz")) {
-            return new PrintStream(new GZIPOutputStream(new FileOutputStream(file)));
+    public static PrintStream makePrintStreamMaybeGzipped(GATKPath filePath) throws IOException {
+        if (filePath.hasExtension(".gz")) {
+            return new PrintStream(new GZIPOutputStream(filePath.getOutputStream()));
         } else {
-            return new PrintStream(file);
+            return new PrintStream(filePath.getOutputStream());
```

Comment:
Will this constructor throw `IOException` if there's an error, or suppress the error?

Reply:
Suppress, same as before. I'm adding a local `checkError` call to this call site. I've added a separate bullet point to https://github.com/broadinstitute/gatk/issues/6610 to replace all usages of `PrintStream` with `OutputStreamWriter`.